handler/core/src/main/java/io/knotx/fragments/handler/ExecutionPlan.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.handler;
+
+import io.knotx.fragments.api.Fragment;
+import io.knotx.fragments.engine.*;
+import io.knotx.fragments.engine.api.Task;
+import io.knotx.server.api.context.ClientRequest;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ExecutionPlan {
+
+  TaskProvider taskProvider;
+  Map<FragmentEventContext, TaskWithMetadata> plan;
+
+  ExecutionPlan(List<Fragment> fragments, ClientRequest clientRequest, TaskProvider taskProvider) {

```

Comment:
This class should be data structure class, however, it contains some logic about creating tasks.